src/pages/iou/request/step/IOURequestStepMerchant.js

```diff
@@ -26,29 +30,42 @@ const propTypes = {
     /** Onyx Props */
     /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
     transaction: transactionPropTypes,
+
+    /** The draft transaction that holds data to be persisted on the current transaction */
+    splitDraftTransaction: transactionPropTypes,
 };
 
 const defaultProps = {
     transaction: {},
+    splitDraftTransaction: {},
 };
 
 function IOURequestStepMerchant({
     route: {
-        params: {transactionID, backTo},
+        params: {transactionID, reportID, backTo, action, iouType},
     },
-    transaction: {merchant, participants},
+    transaction,
+    splitDraftTransaction,
 }) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {inputCallbackRef} = useAutoFocusInput();
+
+    // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
+    const isEditingSplitBill = iouType === CONST.IOU.TYPE.SPLIT && action === CONST.IOU.ACTION.EDIT;
+
+    const {merchant} = ReportUtils.getTransactionDetails(isEditingSplitBill && !lodashIsEmpty(splitDraftTransaction) ? splitDraftTransaction : transaction);
+
     const isEmptyMerchant = merchant === '' || merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
 
-    const isMerchantRequired = _.some(participants, (participant) => Boolean(participant.isPolicyExpenseChat));
+    const isMerchantRequired = _.some(transaction.participants, (participant) => Boolean(participant.isPolicyExpenseChat));
 
     const navigateBack = () => {
         Navigation.goBack(backTo || ROUTES.HOME);
     };
 
+    const isEditing = action === CONST.IOU.ACTION.EDIT;
```

Comment:
Since `action === CONST.IOU.ACTION.EDIT;` is duplicated, can you define this above `isEditingSplitBill` and use it in the definition of `isEditingSplitBill`, ie, `const isEditingSplitBill = iouType === CONST.IOU.TYPE.SPLIT && isEditing`?